v2/tools/asoctl/internal/crd/cleaner.go

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+package crd
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+
+	apiextensionsV1 "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset/typed/apiextensions/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func CleanDeprecatedCRDVersions(ctx context.Context, cl apiextensionsV1.CustomResourceDefinitionInterface) error {
+	var crdRegexp = regexp.MustCompile(".*\\.azure\\.com")
+	var deprecatedVersionRegexp = regexp.MustCompile("v1alpha1api.{8}.*")
+
+	list, err := cl.List(ctx, v1.ListOptions{})
+	if err != nil {
+		return err
+	}
+
+	if list.Items == nil || len(list.Items) == 0 {
```

Comment:
minor: Possibly `nil` deref here, `list == nil`? IMO change to

```suggestion
	if list == nil || len(list.Items) == 0 {
```